### PR TITLE
[CSB-1114] Update with new platform BOM

### PR DESCRIPTION
**Pull request by smongiar**

### CODE REVIEW

**Comment by smongiar** [comment]
@Croway could you review please?

File: README.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ mvn archetype:generate \
 -Dversion=0.0.1 \
 -Dpackage=org.apache.camel.archetypes.archetypeIT \
 -Dmaven-compiler-plugin-version=3.8.1 \
--Dspring-boot-version=2.6.2 \
--Dcamel-version=3.14.0 \
+-Dspring-boot-version=2.7.8 \
+-Dcamel-version=3.20.1 \
```
Comment (Croway) [inline README.md:16]: can these be removed?

File: pom.xml [hunk 2]
```diff
@@ -76,11 +76,23 @@
 		<maven.javadoc.plugin.version>3.4.0</maven.javadoc.plugin.version>
 		<nexus.staging.plugin.version>1.6.13</nexus.staging.plugin.version>
 		<maven.gpg.plugin.version>3.0.1</maven.gpg.plugin.version>
-		<spring-boot-version>2.6.2</spring-boot-version>
+		<spring-boot-version>2.7.8</spring-boot-version>
```
Comment (Croway) [inline pom.xml:79]: can this be removed?

File: src/main/resources-filtered/META-INF/maven/archetype-metadata.xml [hunk 3]
```diff
@@ -27,6 +27,9 @@
     <requiredProperty key="spring-boot-version">
```
Comment (Croway) [inline src/main/resources-filtered/META-INF/maven/archetype-metadata.xml:24]: same here, I don't think we need to specify spring-boot-version anymore

File: src/test/resources/projects/build-it/archetype.properties [hunk 4]
```diff
@@ -25,4 +25,5 @@ logback-version=${logback-version}
 maven-compiler-plugin-version=3.8.1
 maven-resources-plugin-version=${maven-resources-plugin-version}
 spring-boot-version=${spring-boot-version}
-camel-version=3.14.0
+camel-version=3.20.1
```
Comment (Croway) [inline src/test/resources/projects/build-it/archetype.properties]: is it needed?

File: src/test/resources/projects/run-it/archetype.properties [hunk 5]
```diff
@@ -24,4 +24,5 @@ logback-version=${logback-version}
 maven-compiler-plugin-version=3.8.1
 maven-resources-plugin-version=${maven-resources-plugin-version}
 spring-boot-version=${spring-boot-version}
-camel-version=3.14.0
+camel-version=3.20.1
```
Comment (Croway) [inline src/test/resources/projects/run-it/archetype.properties]: needed?

File: src/test/resources/projects/build-it/archetype.properties (hunk 4 above)
Comment (smongiar) [reply src/test/resources/projects/build-it/archetype.properties]: AFAIK those properties are useful for default values definition in combination with archetype-metadata above. I.e.: you can run:
`mvn archetype:generate -DarchetypeGroupId=software.tnb -DarchetypeArtifactId=camel3-archetype-spring-boot -DarchetypeVersion=0.1.4 -DgroupId=test.group -DartifactId=build-it -Dversion=0.0.1 -Dpackage=org.apache.camel.archetypes.archetypeIT -Dmaven-compiler-plugin-version=3.8.1 -B`

File: src/main/resources-filtered/META-INF/maven/archetype-metadata.xml (hunk 3 above)
Comment (smongiar) [reply src/main/resources-filtered/META-INF/maven/archetype-metadata.xml:24]: "_required_" doesn't mean that the user should necessarily specify the parameter. That's managed from a default value, but obviously, we can delete

File: src/test/resources/projects/run-it/archetype.properties (hunk 5 above)
Comment (smongiar) [reply src/test/resources/projects/run-it/archetype.properties]: same here

**Review by Croway — APPROVED** [review]

**Pull request merged**